src/langchain_google_alloydb_pg/engine.py

```diff
@@ -604,3 +605,268 @@ async def _aload_table_schema(
             )
 
         return metadata.tables[table_name]
+
+    def _get_collection_uuid(
+        self,
+        collection_name: str,
+        pg_collection_table_name: Optional[str] = "langchain_pg_collection",
+    ) -> str:
+        """
+        Get the collection uuid for a collection present in PGVector tables.
+
+        Args:
+            collection_name (str): The name of the collection to get the uuid for.
+            pg_collection_table_name (str): The table name which stores the collection uuid to name mappings.
+                Default: "langchain_pg_collection". Optional.
+
+        Returns:
+            The uuid corresponding to the collection.
+        """
+        try:
+            collection_info = self._fetch(
```

Comment:
The upcoming refactor we will be deleting fetch/execute. Any where that you use fetch or execute use the async pool directly like `async with self.engine.connect() as conn:...`. Then we will wrap the async methods with sync wrappers. For example extract_pgvector_collection could be _aextract_pgvector_collection. Then in extract_pgvector_collection we will use run_as_sync to run the _aextract_pgvector_collection methods. See https://github.com/googleapis/langchain-google-cloud-sql-pg-python/blob/c3a0f12276eef07a64ea40b20abe496496b5132c/src/langchain_google_cloud_sql_pg/engine.py#L364 for an example

Reply:
I'll refactor this PR once the async/sync refactor is pushed.